docs/painless/painless-contexts/painless-watcher-condition-context.asciidoc

```diff
@@ -1,38 +1,146 @@
+to determine execution of said action.
 
-*Variables*
+*Return*
 
-`params` (`Map`, read-only)::
-        User-defined parameters passed in as part of the query.
+`boolean`::
+        Expects `true` if the condition is met, and `false` otherwise.
 
-`ctx['watch_id']` (`String`, read-only)::
-        The id of the watch.
+*API*
 
-`ctx['execution_time']` (`ZonedDateTime`, read-only)::
-        The start time for the watch.
+The standard <<painless-api-reference, Painless API>> is available.
 
-`ctx['trigger']['scheduled_time']` (`ZonedDateTime`, read-only)::
-        The scheduled trigger time for the watch.
+The first example is a condition for the watch itself. This determines if any
+of the actions in the watch are executed. The example aggregates the total sold
+seats for each play in the data set. The condition validates that there is at
+least one play that has sold under $15,000 or over $50,000.
 
-`ctx['trigger']['triggered_time']` (`ZonedDateTime`, read-only)::
-        The actual trigger time for the watch.
+[source,Painless]
+----
+POST _watcher/watch/_execute
+{
+  "watch" : {
+    "trigger" : { "schedule" : { "interval" : "24h" } },
+    "input" : {
+      "search" : {
+        "request" : {
+          "indices" : [ "seats" ],
+          "body" : {
+            "query" : {
+              "term": { "sold": "true"}
+            },
+            "aggs" : {
+              "theatres" : {
+                "terms" : { "field" : "play" },
+                "aggs" : {
+                  "money" : {
+                    "sum": { "field" : "cost" }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    },
+    "condition" : {
+      "script" :
+      """
+        return ctx.payload.aggregations.theatres.buckets.stream()       <1>
+          .filter(t -> t.money.value < 15000 || t.money.value > 50000)  <2>
+          .collect(Collectors.toList())                                 <3>
+          .size() > 0                                                   <4>
+      """
+    },
+    "actions" : {
+      "my_log" : {
+        "logging" : {
+          "text" : "The output of the search was : {{ctx.payload.aggregations.theatres.buckets}}"
+        }
+      }
+    }
+  }
+}
+----
 
-`ctx['metadata']` (`Map`, read-only)::
-        Any metadata associated with the watch.
+<1> The Java Stream API is used in the condition. This API allows manipulation of
+the elements of the list in a pipeline.
+<2> A Stream Filter is used to remove items that do not meet the criteria within the
+filter.
+<3> The collector reduces the stream back into a java.util.List
+<4> The size of the list is is validated, and if it returns greater than zero entries,
+the condition passes and is executed.
 
-`ctx['payload']` (`Map`, read-only)::
-        The accessible watch data based upon the
-        {xpack-ref}/input.html[watch input].
+==== Action condition
 
```

Comment:
I'd remove the heading and just intro the second example (see following comment).